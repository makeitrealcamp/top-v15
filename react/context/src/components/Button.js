// import { useContext } from 'react'
// import { CounterContext } from '../context/counter'
import { useCounter } from '../utils/useCounter'

export function Button() {
  // const { setCount } = useContext(CounterContext);

  const { setCount } = useCounter();

  return (
    <div>
      <button
        type="button"
        onClick={() => setCount(prevCount => prevCount + 1)}
      >
        Increment
      </button>
    </div>
  )
}

