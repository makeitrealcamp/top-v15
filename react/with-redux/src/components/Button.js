// import { connect } from 'react-redux';
import { useDispatch } from 'react-redux';
import { INCREMENT, DECREMENT } from '../store';

export function Button() {
  const dispatch = useDispatch();

  return (
    <div>
      <button
        type="button"
        onClick={() => dispatch({ type: INCREMENT })}
      >
        Increment
      </button>
      <button
        type="button"
        onClick={() => dispatch({ type: DECREMENT })}
      >
        Decrement
      </button>
    </div>
  )
}

// function mapStateToProps(state) {
//   return {
//     count: state.count,
//   }
  // <Button count={state.count} />
// }

// function mapDispatchToProps(dispatch) {
//   return {
//     increment: () => dispatch({ type: INCREMENT }),
//   }
  // <Button increment={() => dispatch({ type: INCREMENT })} />
// }

// function connect(mapStateToProps, mapDispatchToProps) {
//   const reduxProps = {
//     ...mapStateToProps(store.getState()),
//     ...mapDispatchToProps(store.dispatch),
//   };

//   return function (Component) {
//     return function (props) {
//       return <Component {...props} {...reduxProps} />
//     }
//   }
// }

// export default connect(mapStateToProps, mapDispatchToProps)(Button);
