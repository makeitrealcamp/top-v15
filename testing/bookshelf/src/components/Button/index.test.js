import { render, screen } from '@testing-library/react';
import Button from '.';

describe('Button', () => {
  it('should disable button when content has less than 5 characters', () => {
    render(<Button type="button">hola</Button>);

    const button = screen.getByTestId('button');

    expect(button).toBeDisabled();
  });

  // esta prueba no es tan chÃ©vere
  it('should render the children prop as button content', () => {
    render(<Button type="button">click</Button>);

    const button = screen.getByText(/click/i);

    expect(button).toBeInTheDocument();
  });

  // Snapshot testing
  // No no no no Falsos Positivos y Negativos
  it('should render correctly', () => {
    const { container } = render(<Button type="button">click</Button>);

    expect(container).toMatchSnapshot();
  })
})
