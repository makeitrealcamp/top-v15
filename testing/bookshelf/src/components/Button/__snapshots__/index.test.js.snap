// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly 1`] = `
<div>
  <button
    data-testid="button"
    type="button"
  >
    click
    <div />
  </button>
</div>
`;
